packages/babel-generator/src/buffer.js

```diff
@@ -1,251 +1,92 @@
 import Position from "./position";
-import repeat from "lodash/repeat";
 import trimEnd from "lodash/trimEnd";
 
-/**
- * Buffer for collecting generated output.
- */
-
 export default class Buffer {
-  constructor(format: Object) {
-    this.printedCommentStarts = {};
-    this.parenPushNewlineState = null;
-    this._position = new Position();
-    this._indent = format.indent.base;
-    this.format = format;
-    this.buf = "";
-
-    // Maintaining a reference to the last char in the buffer is an optimization
-    // to make sure that v8 doesn't "flatten" the string more often than needed
-    // see https://github.com/babel/babel/pull/3283 for details.
-    this.last = "";
```

Comment:
I added some documentation at the top of `buffer.js` to explain the performance goal of this class.